### ISSUE TITLE
Can't pair PSMove controller - have tried everything I could think of

### ISSUE THREAD

**Issue by jadtheaussie** [cboulay/PSMoveService#506]
Hi, I've searched for many solutions for my problem but have found that none have worked so I have opened an issue. My PSMove controllers aren't pairing, as said in the title and I don't know why. I'm usiong two PSEye cameras, both working with the service, two PSMove controllers, and a CSR 4.0 bluetooth adapter with Windows default drivers installed.

Here is my PSMoveService log:
`[2018-01-14 13:15:28.641]: main - Starting PSMoveService v0.9-alpha 8.7.2 (protocol v0.9-alpha 8.7.0)
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\USBManagerConfig.json"
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\USBManagerConfig.json"
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\NetworkManagerConfig.json"
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\NetworkManagerConfig.json"
[2018-01-14 13:15:28.654]: USBAsyncRequestManager::startup - Requested LibUSBApi
[2018-01-14 13:15:28.655]: USBAsyncRequestManager::startup - Creating LibUSBApi
[2018-01-14 13:15:28.658]: USBAsyncRequestManager::startup - Initialized USB API
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\DeviceManagerConfig.json"
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\DeviceManagerConfig.json"
[2018-01-14 13:15:28.663]: DeviceManager::startup - Platform Hotplug API is ENABLED
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\ControllerManagerConfig.json"
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\ControllerManagerConfig.json"
[2018-01-14 13:15:28.786]: bluetooth_get_host_address - Found a bluetooth radio
[2018-01-14 13:15:28.789]: bluetooth_get_host_address - Retrieved radio info
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\TrackerManagerConfig.json"
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\TrackerManagerConfig.json"
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\HMDManagerConfig.json"
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\HMDManagerConfig.json"
libusb: error [init_device] device '\\.\USB#VID_1415&PID_2000&MI_01#6&348A1D35&0&0001' is no longer connected!
libusb: error [init_device] device '\\.\USB#VID_1038&PID_1394&MI_03#6&253F2F8E&2&0003' is no longer connected!
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [init_device] device '\\.\USB#VID_1415&PID_2000&MI_01#6&348A1D35&0&0001' is no longer connected!
libusb: error [init_device] device '\\.\USB#VID_1038&PID_1394&MI_03#6&253F2F8E&2&0003' is no longer connected!
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
[2018-01-14 13:15:28.922]: PS3EyeTracker::open - Opening PS3EyeTracker(USB\VID_1415&PID_2000\b1_p9, camera_index=0)
libusb: error [init_device] device '\\.\USB#VID_1415&PID_2000&MI_01#6&348A1D35&0&0001' is no longer connected!
libusb: error [init_device] device '\\.\USB#VID_1038&PID_1394&MI_03#6&253F2F8E&2&0003' is no longer connected!
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
ps3eye::PS3EYECam::getDevices() found 1 devices.
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_ps3eye_b1_p9.json"
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_ps3eye_b1_p9.json"
[2018-01-14 13:15:29.114]: SharedMemory::initialize() - Allocating shared memory: tracker_view_0
[2018-01-14 13:15:29.299]: DeviceTypeManager::update_connected_devices - Device device_id 0 (PSEYE) opened
[2018-01-14 13:15:29.303]: TrackerDeviceEnumerator - Skipping device (USB\VID_1415&PID_2000\b1_p9) - Operation not supported or unimplemented on this platform
[2018-01-14 13:17:20.94]: ClientConnection::start - Starting client connection id 0
[2018-01-14 13:17:20.97]: ClientConnection::send_connection_info - Sending connection id to client 0
libusb: error [init_device] device '\\.\USB#VID_1415&PID_2000&MI_01#6&348A1D35&0&0001' is no longer connected!
libusb: error [init_device] device '\\.\USB#VID_1038&PID_1394&MI_03#6&253F2F8E&2&0003' is no longer connected!
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
[2018-01-14 13:17:27.767]: PSMoveController::open - Opening PSMoveController(\\?\hid#vid_054c&pid_03d5&col01#6&26906822&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030})
[2018-01-14 13:17:27.771]: PSMoveController::open -   with EMPTY serial_number
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\04_76_6e_07_dd_fc.json"
Config file name: "C:\Users\Jadon\AppData\Roaming\PSMoveService\04_76_6e_07_dd_fc.json"
[2018-01-14 13:17:27.780]: DeviceTypeManager::update_connected_devices - Device device_id 0 (PSMove) opened
libusb: error [init_device] device '\\.\USB#VID_1415&PID_2000&MI_01#6&348A1D35&0&0001' is no longer connected!
libusb: error [init_device] device '\\.\USB#VID_1038&PID_1394&MI_03#6&253F2F8E&2&0003' is no longer connected!
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [init_device] device '\\.\USB#VID_1415&PID_2000&MI_01#6&348A1D35&0&0001' is no longer connected!
libusb: error [init_device] device '\\.\USB#VID_1038&PID_1394&MI_03#6&253F2F8E&2&0003' is no longer connected!
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
[2018-01-14 13:17:33.164]: AsyncBluetoothPairDeviceRequest - Found a bluetooth radio
[2018-01-14 13:17:33.167]: AsyncBluetoothPairDeviceRequest - Retrieved radio info
[2018-01-14 13:17:33.272]: AsyncBluetoothPairDeviceRequest - Assigned host address 00:1a:7d:da:71:0b to controller id 0
[2018-01-14 13:17:33.275]: ServerRequestHandler - Async bluetooth request([Pair] ID: 0 Conn: 0) started.
[2018-01-14 13:17:33.276]: AsyncBluetoothPairDeviceRequest - Making radio accept incoming connections
[2018-01-14 13:17:33.283]: AsyncBluetoothPairDeviceRequest - Failed to enable incoming connections on radio 00:1a:7d:da:71:0b
[2018-01-14 13:17:33.286]: AsyncBluetoothPairDeviceRequest - Making radio discoverable
[2018-01-14 13:17:33.289]: AsyncBluetoothPairDeviceRequest - Failed to enable radio 00:1a:7d:da:71:0b discoverable
[2018-01-14 13:17:33.291]: ServerRequestHandler - Async bluetooth request([Pair] ID: 0 Conn: 0) failed!
`
I plug in the controller when I have the config tool open and when I go to pair the controller, it just says it failed to pair. Is there any noticeable problem looking at my log?

**Comment by OpaqueEcho** [comment]
I don't think you can pair the controller via Bluetooth while it's connected via USB.

**Comment by jadtheaussie** [comment]
Exactly what do you mean by that? You have to connect the controller first by mini USB to get the service to recognise it right?

**Comment by jadtheaussie** [comment]
The bluetooth adapter itself is working, I tried and succeeded in connecting my phone to the computer via Bluetooth. In that case, it must be the PSMoveService itself or the controllers, or that my CSR 4.0 doesn't work with my controllers. But I am not an expert, and have no clue which one it is, and what the problem is.

**Comment by lewilou22** [comment]
Ialso have a CSR 4.0 not working with move controllers PLEASE HELP

**Comment by cboulay** [comment]
I tried a few BT adapters and the only one I ever got it to work with was the ASUS BT-400. I'm sure it's chipset dependent and there are probably several chipsets (and many devices) that it works with but I'm sorry I don't have a whitelist for you. It's possible that it doesn't work with your dongle.
If it's a newer PS4 controller then please follow #512

**Comment by 12135555** [comment]
Faced the same error. Here is a fix:

1. Open services.msc through "Run"
2. Find "Bluetooth support service"
3. Open properties of this service -> Run: Automatically
4. Run that service

Worked perfectly!

**Comment by eyeEmotion** [comment]
I'm also trying everything I can find online, but nothing seems to work.
My bluetooth is built-in. Laptop is around 1 and a half year old. My Move-controllers are original ones from the PS3-days.

This is what I get when starting the PSMoveService:

> [2020-04-14 23:57:26.287]: main - Starting PSMoveService v0.9-alpha 9.0.1 (protocol v0.9-alpha 9.0.0)
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\USBManagerConfig.json"
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\USBManagerConfig.json"
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\NetworkManagerConfig.json"
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\NetworkManagerConfig.json"
[2020-04-14 23:57:26.298]: USBAsyncRequestManager::startup - Requested LibUSBApi
[2020-04-14 23:57:26.300]: USBAsyncRequestManager::startup - Creating LibUSBApi
[2020-04-14 23:57:26.304]: USBAsyncRequestManager::startup - Initialized USB API
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\DeviceManagerConfig.json"
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\DeviceManagerConfig.json"
[2020-04-14 23:57:26.311]: DeviceManager::startup - Platform Hotplug API is ENABLED
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\ControllerManagerConfig.json"
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\ControllerManagerConfig.json"
[2020-04-14 23:57:26.355]: bluetooth_get_host_address - Found a bluetooth radio
[2020-04-14 23:57:26.358]: bluetooth_get_host_address - Retrieved radio info
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\TrackerManagerConfig.json"
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\TrackerManagerConfig.json"
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\HMDManagerConfig.json"
Config file name: "C:\Users\eyeEmotion\AppData\Roaming\PSMoveService\HMDManagerConfig.json"
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent

When I try to pair it with the ConfigurationTool, I can see some of these messages:

> [2020-04-15 00:01:34.759]: AsyncBluetoothPairDeviceRequest - Connection attempt: 53/80
[2020-04-15 00:01:34.763]: AsyncBluetoothPairDeviceRequest - Device not connected
[2020-04-15 00:01:35.66]: AsyncBluetoothPairDeviceRequest - Connection attempt: 54/80
[2020-04-15 00:01:35.70]: AsyncBluetoothPairDeviceRequest - Patching the registry ...
[2020-04-15 00:01:35.72]: AsyncBluetoothPairDeviceRequest - Failed to open registry key, it does not yet exist
[2020-04-15 00:01:35.74]: AsyncBluetoothPairDeviceRequest - Checking HID service
[2020-04-15 00:01:35.77]: AsyncBluetoothPairDeviceRequest - HID service not enabled, attempting to enable
[2020-04-15 00:01:38.94]: AsyncBluetoothPairDeviceRequest - Patching the registry ...
[2020-04-15 00:01:38.97]: AsyncBluetoothPairDeviceRequest - Failed to open registry key, it does not yet exist
[2020-04-15 00:01:38.99]: AsyncBluetoothPairDeviceRequest - Verification attempt 0 / 5
[2020-04-15 00:01:38.102]: AsyncBluetoothPairDeviceRequest - Device Remembered
[2020-04-15 00:01:38.104]: AsyncBluetoothPairDeviceRequest - HID service enabled
[2020-04-15 00:01:38.106]: AsyncBluetoothPairDeviceRequest - HID service not enabled
[2020-04-15 00:01:38.109]: AsyncBluetoothPairDeviceRequest - Verified failed. Re-establish connection
[2020-04-15 00:01:38.111]: AsyncBluetoothPairDeviceRequest - Connection attempt: 0/80
[2020-04-15 00:01:38.115]: AsyncBluetoothPairDeviceRequest - Device not connected
[2020-04-15 00:01:38.417]: AsyncBluetoothPairDeviceRequest - Connection attempt: 1/80
[2020-04-15 00:01:38.421]: AsyncBluetoothPairDeviceRequest - Device not connected

Funny thing is that it says my HID service isn't enabled, but I checked that and it is enabled. I set it to start automatically. Also tried the Generic Bluetooth Adapter, but that didn't work either.

Need some help, not really tech-savvy on that part.
